File: jna/src/main/java/dark/bbox_t.java
package dark;

import com.sun.jna.Pointer;
import com.sun.jna.Structure;

import java.util.Arrays;
import java.util.List;
/*
struct bbox_t {
    unsigned int x, y, w, h;       // (x,y) - top-left corner, (w, h) - width & height of bounded box
    float prob;                    // confidence - probability that the object was found correctly
    unsigned int obj_id;           // class of object - from range [0, classes-1]
    unsigned int track_id;         // tracking id for video (0 - untracked, 1 - inf - tracked object)
    unsigned int frames_counter;   // counter of frames on which the object was detected
    float x_3d, y_3d, z_3d;        // center of object (in Meters) if ZED 3D Camera is used
};
 */
public class bbox_t extends Structure implements Structure.ByValue {
    public int x, y, w, h;       // (x,y) - top-left corner, (w, h) - width & height of bounded box
    public float prob;                    // confidence - probability that the object was found correctly
    public int obj_id;           // class of object - from range [0, classes-1]
    public int track_id;         // tracking id for video (0 - untracked, 1 - inf - tracked object)
    public int frames_counter;   // counter of frames on which the object was detected
    public float x_3d, y_3d, z_3d;        // center of object (in Meters) if ZED 3D Camera is used


    public bbox_t() {
        super();
    }

    public bbox_t(Pointer peer) {
        super(peer);
    }

    @Override
    protected List<String> getFieldOrder() {
        return Arrays.asList("x","y","w","h","prob","obj_id","track_id","frames_counter","x_3d","y_3d","z_3d");
    }
}

File: jna/src/main/java/dark_kotlin/bbox_t.kt
package com.sl.microservice.ai.runtime.impl.darknet.dark

import com.sun.jna.Pointer
import com.sun.jna.Structure


class bbox_t : Structure, Structure.ByValue {
    @JvmField val x = null
    @JvmField val y = null
    @JvmField val w = null
    @JvmField val h = null  // (x,y) - top-left corner, (w, h) - width & height of bounded box
    @JvmField val prob = 0f // confidence - probability that the object was found correctly
    @JvmField val obj_id = 0 // class of object - from range [0, classes-1]
    @JvmField val track_id = 0 // tracking id for video (0 - untracked, 1 - inf - tracked object)
    @JvmField val frames_counter = 0// counter of frames on which the object was detected
    @JvmField val x_3d = 0f
    @JvmField val y_3d = 0f
    @JvmField val z_3d = 0f // center of object (in Meters) if ZED 3D Camera is used

    constructor() : super()
    constructor(peer: Pointer) : super(peer)

    override fun getFieldOrder(): List<String> {
        return listOf("x", "y", "w", "h", "prob", "obj_id", "track_id", "frames_counter", "x_3d", "y_3d", "z_3d")
    }
}
